DSharpPlus.CommandsNext/Converters/DefaultHelpFormatter.cs

```diff
@@ -98,7 +98,21 @@ public override BaseHelpFormatter WithCommand(Command command)
         /// <returns>This help formatter.</returns>
         public override BaseHelpFormatter WithSubcommands(IEnumerable<Command> subcommands)
         {
-            this.EmbedBuilder.AddField(this.Command is not null ? "Subcommands" : "Commands", string.Join(", ", subcommands.Select(x => Formatter.InlineCode(x.Name))), false);
+            var categories = subcommands.Where(xm => xm.Category != null).Select(xm => xm.Category).Distinct();
+
+            if (categories.Count() == 0)
+            {
+                this.EmbedBuilder.AddField(this.Command is not null ? "Subcommands" : "Commands", string.Join(", ", subcommands.Select(x => Formatter.InlineCode(x.Name))), false);
+
+                return this;
+            }
+
+            foreach(var category in categories)
+            {
+                this.EmbedBuilder.AddField(category, string.Join(", ", subcommands.Where(xm => xm.Category == category).Select(xm => Formatter.InlineCode(xm.Name))), false);
```

Comment:
Instead of constantly re-iterating over the entire command list, consider a `GroupBy` statement instead.